optimum/exporters/openvino/model_patcher.py

```diff
@@ -497,50 +497,65 @@ def _llama_gemma_update_causal_mask_latest(
     _llama_gemma_update_causal_mask = _llama_gemma_update_causal_mask_legacy
 
 
-class GemmaModelPatcher(DecoderModelPatcher):
-    def __enter__(self):
-        super().__enter__()
-
-        # gemma has some accuracy issues with bf16 with transformers >= 4.39
-        # fill causal mask in slightly different way for avoid overflow on some platforms
-        if is_transformers_version(">=", "4.39.0"):
-            self._model.model._orig_update_causal_mask = self._model.model._update_causal_mask
-            self._model.model._update_causal_mask = types.MethodType(
-                _llama_gemma_update_causal_mask, self._model.model
-            )
-
-        # init inv_freq for torchscript tracing
-        # https://github.com/huggingface/transformers/blob/ed74d97871468f3a4695ede50abdc0b55717a84d/src/transformers/models/gemma/modeling_gemma.py#L108
-        for layer in self._model.model.layers:
-            if layer.self_attn.rotary_emb.inv_freq is None:
-                rotary_emb = layer.self_attn.rotary_emb
-                layer.self_attn.rotary_emb.inv_freq = 1.0 / (
-                    rotary_emb.base ** (torch.arange(0, rotary_emb.dim, 2, dtype=torch.int64).float() / rotary_emb.dim)
-                )
-
-    def __exit__(self, exc_type, exc_value, traceback):
-        super().__exit__(exc_type, exc_value, traceback)
-        if hasattr(self._model.model, "_orig_update_causal_mask"):
-            self._model.model._update_causal_mask = self._model.model._orig_update_causal_mask
+def llama_gemma_rotary_emb_forward(self, x, position_ids, seq_len=None):
+    _seq_len = torch.max(position_ids) + 1 if seq_len is None else seq_len
+    if _seq_len > self.embed_positions.shape[0]:
+        if seq_len is None:
+            return self._orig_forward(self, x, position_ids)
+        else:
+            return self._orig_forward(self, x, position_ids, seq_len)
```

Comment:
where does the `seq_len` argument comes from ?  

https://github.com/huggingface/transformers/blob/v4.42.3/src/transformers/models/llama/modeling_llama.py#L107

```suggestion
        return self._orig_forward(x, position_ids)
```

Reply:
from gemma https://github.com/huggingface/transformers/blob/v4.42.3/src/transformers/models/gemma/modeling_gemma.py#L104

Reply:
This will fail for llama models then no ? I think we need to have a test for this

Reply:
not sure that I understand you... if llama does not have seq_len argument, then it will be not provided to forward and it is equal to None as default argument, so passing seq_len to original forward will never happens for it

Reply:
if you speak what happens if current sequence len > current cached (max position embeddings 2048 for llama2, 4096 for llama3 if I right remember), it is expected that model will fail due to tracing limitations and the same for onnx and other models as well. For pytorch if model is not trained to extend max context, it will continue generation but as the rule it will generate some kind of trash  (random tokens), so anyway after that results will be nonvalid.

We are working for supporting models with context extension, but not ready to commit on that right now